src/main/java/org/spongepowered/api/service/permission/SubjectCollection.java

```diff
@@ -24,86 +24,315 @@
+     * data is not considered.</p>
      *
-     * <p>Note: This data may be persisted, so plugins that add permissions to
-     * this subject must take care to not override permissions already set or
-     * modified. It is also recommended to use
-     * {@link Subject#getTransientSubjectData()} where possible to
-     * avoid persisting unnecessary data.
+     * <p>As no context is passed, it is up to the implementation to decide
+     * which contexts to use. When available,
+     * {@link Subject#getActiveContexts()} is used for the lookup. Otherwise, it
+     * is likely that {@link SubjectData#GLOBAL_CONTEXT} will be
+     * used.</p>
      *
-     * @return The subject holding defaults
+     * @param permission The permission to check
+     * @return A map containing any subject known to have this permission set,
+     *         and the value this permission is set to
+     */
+    Map<Subject, Boolean> getLoadedWithPermission(String permission);
+
+    /**
+     * Return all loaded subjects with the given permission set.
+     *
+     * <p>This method <p>DOES NOT</p> consider inheritance, and will only query
+     * the data set to the subjects {@link Subject#getSubjectData()}. Transient
+     * data is not considered.</p>
+     *
+     * @param contexts The context combination to check for permissions in
+     * @param permission The permission to check
+     * @return A map containing any subject known to have this permission set,
+     *         and the value this permission is set to
+     */
+    Map<Subject, Boolean> getLoadedWithPermission(Set<Context> contexts, String permission);
+
+    /**
+     * Gets the subject holding data that is applied by default to all
+     * subjects in this collection.
+     *
+     * <p>This subject is at the root of all inheritance trees for subjects in
+     * this collection, but at a higher priority chan defaults expressed to
+     * {@link PermissionService#getDefaults()}.</p>
+     *
+     * <p>Note: This data may be persisted, so plugins that add
+     * permissions to this subject must take care to not override
+     * permissions already set or modified.</p>
+     *
+     * <p>It is also recommended to use
+     * {@link Subject#getTransientSubjectData()} where possible to avoid
+     * persisting unnecessary data.</p>
+     *
+     * <p>Assigning default permissions should be used sparingly, and by
+     * convention, only in situations where "default" game behaviour is restored
+     * by granting a certain permission.</p>
+     *
+     * @return The subject holding defaults for this collection
      */
     Subject getDefaults();
+
+    /**
+     * Indicate that a certain subject may be unloaded by the implementation.
+     *
+     * <p>This is only a hint to the implementation, and does not guarantee that
+     * the subject will be unloaded.</p>
+     *
+     * @param identifier The subject to be unloaded
+     */
+    void suggestUnload(String identifier);
```

Comment:
The consequence of the method is intentionally left open. (to make it flexible for implementations, this is really important in the PS API.)

Some possible results:

* Implementation just ignores the call (returns false??)
* Implementation always listens to the call, and unloads the subject no matter what. (returns true)
* Implementation normally listens to the call, but for a certain subject, doesn't want to unload, say, if the related user was online. (returns false?)

Lots of different possible outcomes, with very vague information being presented to the user, which I would argue, has no use, and couldn't really be depended on.

Idk, like:
```java
SubjectCollection collection;
Subject subj;

boolean ret = collection.suggestUnload(subj.getIdentifier());

// then do what with the ret? how would it ever be useful!?
// even if ret was true/false, what does that actually mean?
```

I think that by adding a return, you might encourage users to act upon the result, which would be totally arbitrary.